src/pip/_internal/operations/install/wheel.py

```diff
@@ -639,6 +641,14 @@ def _generate_file(path, **kwargs):
         installer_file.write(b'pip\n')
     generated.append(installer_path)
 
+    # Document source package hash
+    hash_file_path = os.path.join(dest_info_dir, 'HASH')
+    with open(hash_file_path, 'w') as dest_hash_file:
+        for alg in STRONG_HASHES:
+            dest_hash_file.write('{}:{}\n'.format(
+                alg, hash_of_file(cast('str', wheel_zip.filename), alg)))
+    generated.append(hash_file_path)
```

Comment:
I see, I will be happy to do so, although I have no idea how to format any `PEP` :smile:

Reply:
I am not too familiar with that part too,  But I would way for the maintainers to share their thoughts on this.

Reply:
I do not think we need to store this information as it is only relevant to installed packages. When package is in the cache it will not be shown in `pip freeze` unless reinstalled